doc/tutorials/basic/build_chatbot.md

```diff
@@ -0,0 +1,148 @@
+# Build a Chat Bot
+
+In this tutorial we will build a streaming *chat bot*. We will first use the *high-level* [`ChatInterface`](../../reference/chat/ChatInterface.ipynb) to build a basic chat bot. Then will will add streaming and finally we will make it scale to many users.
+
+:::{admonition} Note
+When I ask you to *run the code* in the sections below, you may either execute the code directly in the Panel docs via the green *run* button, in a cell in a notebook or in a file `app.py` that is served with `panel serve app.py --autoreload`.
+:::
+
+## Build a Basic Chat Bot
+
+Run the code below:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+
+pn.extension()
+
+WIND_TURBINE = "A *wind turbine* is a device that converts the kinetic energy of wind into electrical energy."
+DONT_KNOW = "Sorry. I don't know."
+
+def get_response(contents, user, instance):
+    if "turbine" in contents.lower():
+        response = WIND_TURBINE
+    else:
+        response = DONT_KNOW
+    sleep(1)
+    return response
+
+chat_bot = pn.chat.ChatInterface(callback=get_response, max_height=500)
+chat_bot.send("Ask me what a wind turbine is", user="Assistant", respond=False)
+chat_bot.servable()
+```
+
+Try entering `What is a wind turbine?` in the *text input* and click *Send*.
+
+:::{admonition}
+The `callback` function `get_response` will receive `"What is a wind turbine?"` in the `contents` argument. Since `contents` contains the word *turbine*, the chat bot will return `WIND_TURBINE` as the `response`.
+:::
+
+## Add Streaming
+
+We will now make the chat bot *stream* its response just like chatGPT does.
+
+Run the code below:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+
+pn.extension()
+
+WIND_TURBINE = "A *wind turbine* is a device that converts the kinetic energy of wind into electrical energy."
+DONT_KNOW = "Sorry. I don't know."
+
+def get_response(contents, user, instance):
+    if "turbine" in contents.lower():
+        response = WIND_TURBINE.split(" ")
+    else:
+        response = DONT_KNOW.split(" ")
```

Comment:
```suggestion
        response = WIND_TURBINE
    else:
        response = DONT_KNOW
```

Reply:
The `.split` is needed to stream tokens back. Otherwise you stream characters back. The first is what happens in the real world. But the .split`. Adds more complexity.

Not sure what to do here?